Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,253 @@
+    files = []
+    found = False
+    for blob in blobs:
+        if blob.name.startswith(lock_repository_name):
+            found = True
+            if blob.name.startswith(f'{lock_repository_name}/{machines_lock_repo}'):
+                lock_file_name = blob.name.split('/')[-1]
+                if lock_file_name:
+                    files.append({'machine_name': lock_file_name.split('-lock-')[0],
+                                  'job_id': lock_file_name.split('-lock-')[1]})
+        elif found:
+            break
+    return files
+
+
+def check_job_status(token: str, job_id: str):
+    """
+    get the status of a job in gitlab.
+    Args:
+        token(str): the gitlab token.
+        job_id(str):  the job id to check.
+    Returns: the status of the job.
+
+    """
+    user_endpoint = f"https://code.pan.run/api/v4/projects/2596/jobs/{job_id}"  # disable-secrets-detection
+    headers = {'PRIVATE-TOKEN': token}
+    response = requests.get(user_endpoint, headers=headers)
+    return response.json().get('status')
+
+
+def remove_build_from_queue(storage_bucket: any, lock_repository_name: str, job_id: str):
+    """deletes a lock queue file """
+    file_path = f'{lock_repository_name}/{QUEUE_REPO}/{job_id}'
+    blob = storage_bucket.blob(file_path)
+    try:
+        blob.delete()
+    except Exception as err:
+        logging.info(f'when we try to delete a build_from_queue = {file_path}, we get an error: {str(err)}')
+        pass
+
+
+def remove_machine_lock_file(storage_bucket: any, lock_repository_name: str, machine_name: str, job_id: str):
+    """deletes a lock machine file """
+    file_path = f'{lock_repository_name}/{machine_name}-lock-{job_id}'
+    blob = storage_bucket.blob(file_path)
+    try:
+        blob.delete()
+    except Exception as err:
+        logging.info(f'when we try to delete a lock machine file = {file_path}, we get an error: {str(err)}')
+        pass
+
+
+def lock_machine(storage_bucket: any, lock_repository_name: str, machine_name: str, job_id: str):
+    """create a lock machine file """
+    blob = storage_bucket.blob(f'{lock_repository_name}/{MACHINES_LOCKS_REPO}/{machine_name}-lock-{job_id}')
+    blob.upload_from_string('')
+
+
+def adding_build_to_the_queue(storage_bucket: any, lock_repository_name: str, job_id: str):
+    """create a lock machine file """
+    blob = storage_bucket.blob(f'{lock_repository_name}/{QUEUE_REPO}/{job_id}')
+    blob.upload_from_string('')
+
+
+def get_my_place_in_the_queue(storage_client: storage.Client, gcs_locks_path: str, job_id: str):
+    """
+    get_my_place_in_the_queue
+    Args:
+        storage_client(storage.Client): The GCP storage client.
+        gcs_locks_path(str): the lock repository name.
+        job_id(str): the job id to check.
+
+    Returns: the place in the queue.
+
+    """
+    logging.info('gets all builds in the queue')
+    builds_in_queue = get_queue_locks_details(storage_client, 'xsoar-ci-artifacts', f'{gcs_locks_path}/{QUEUE_REPO}/')
```

Comment:
I think it is better practice to use the parameters of the functions
```suggestion
    builds_in_queue = get_queue_locks_details(storage_client=storage_client, bucket_name='xsoar-ci-artifacts', prefix=f'{gcs_locks_path}/{QUEUE_REPO}/')
```